pyas2lib/as2.py

```diff
@@ -605,14 +630,17 @@ def parse(self, raw_content, find_org_cb, find_partner_cb, find_message_cb=None)
         try:
             # Get the organization and partner for this transmission
             org_id = unquote_as2name(as2_headers["as2-to"])
-            self.receiver = find_org_cb(org_id)
-            if not self.receiver:
-                raise PartnerNotFound(f"Unknown AS2 organization with id {org_id}")
-
             partner_id = unquote_as2name(as2_headers["as2-from"])
-            self.sender = find_partner_cb(partner_id)
-            if not self.sender:
-                raise PartnerNotFound(f"Unknown AS2 partner with id {partner_id}")
+            if find_org_partner_cb:
+                self.receiver, self.sender = find_org_partner_cb(org_id, partner_id)
```

Comment:
do we not need raise PartnerNotFound here as well?

Reply:
Move the checks down and outside the if statement.